docs/sources/operators-guide/configure/configure-helm-ha-deduplication-consul/index.md

```diff
@@ -0,0 +1,161 @@
+
+You need to have a Grafana Mimir installed via the mimir-distributed Helm chart.
+
+For conceptual information about how Mimir deduplicates incoming HA samples, refer to [Configure high availability]({{< relref "../configuring-high-availability-deduplication.md" >}}).
+
+You also need to configure HA for Prometheus or Grafana Agent. Lastly, you need a key-value store such as Consul KV.
+
+## Configure Prometheus or Grafana Agent to send HA external labels
+
+Configure the Prometheus or Grafana Agent HA setup by setting the labels named `cluster` and `__replica__`,
+which are the default labels for a HA setup in Grafana Mimir. If you want to change the HA labels,
+make sure to change them in Mimir as well. This ensures that the configurations of Grafana Mimir, Prometheus, and Grafana Agent all match each other. Otherwise, HA deduplication will not work.
+
+- The value of the `cluster` label must be same across replica that belong to the same cluster.
+- The value of the `__replica__` label must be unique across different replica within the same cluster.
+
+```yaml
+global:
+  external_labels:
+    __replica__: replica-1
+    cluster: my-prometheus
+```
+
+Reload or restart Prometheus or Grafana Agent after you update the configuration.
+
+> **Note:** [Configure high availability]({{< relref "../configuring-high-availability-deduplication.md" >}}).
+> document contains the same information on Prometheus setup for HA dedup.
+
+## Install Consul using Helm
+
+To get and install Consul on Kubernetes, use the [Consul helm chart](https://github.com/hashicorp/consul-k8s/tree/main/charts/consul).
+Make a note of the Consul endpoint, because you will need it later to configure Mimir.
+
+## Configure Mimir high availability
+
+You can configure Mimir HA deduplication globally or at the tenant level.
+
+### Globally configure HA deduplication
+
+Before you begin, make sure that Mimir was installed using the mimir-distributed Helm chart.
+
+1. Add the following configuration to your `custom.yaml` file:
+
+```yaml
+mimir:
+  structuredConfig:
+    limits:
+      accept_ha_samples: true
+      # The following two configurations must match those of external_labels in Prometheus
+      # The config values below are the default and can be removed if you don't want to override to a new value
+      ha_cluster_label: cluster
+      ha_replica_label: __replica__
+    distributor:
+      ha_tracker:
+        enable_ha_tracker: true
+        kvstore:
+          store: consul
+          consul:
+            host: <consul-endpoint> # example: http://consul.consul.svc.cluster.local:8500
+```
+
+2. Upgrade the Mimir's helm release using the following command:
+
+```bash
+ helm -n <mimir-namespace> upgrade mimir grafana/mimir-distributed -f custom.yaml
+```
+
+### Configure HA deduplication per tenant
+
+Before you begin, make sure that Mimir was installed using the mimir-distributed Helm chart.
+
+1. Add the following configuration to the `custom.yaml` file:
```

Comment:
nit: technically you need to merge (not add) these into your existing custom.yaml, since helm only takes the last mimir.structuredConfig into account otherwise.

Reply:
Nit updated